src/controllers/message/services/MessageServiceFactory.ts

```diff
@@ -0,0 +1,22 @@
+import { ConnectionRecord } from '@credo-ts/core'
+import { InjectQueue } from '@nestjs/bull'
+import { Injectable, Optional } from '@nestjs/common'
+import { Queue } from 'bull'
+
+import { IBaseMessage } from '../../../model'
+
+import { CoreMessageService } from './CoreMessageService'
+
+@Injectable()
+export class MessageServiceFactory {
+  constructor(
+    @Optional() @InjectQueue('message') private messageQueue: Queue,
+    private readonly coreMessageService: CoreMessageService,
+  ) {}
+
+  async setProcessMessage(redisAvailable: boolean, message: IBaseMessage, connection: ConnectionRecord) {
```

Comment:
Why don't call this simply `processMessage`, receiving the message and connection and doing the isRedisAvailable check right here?

Reply:
fixed